### PR TITLE
Add option to configure log level of WebApplicationException

### LINKED ISSUE #31980: Configure RESTEasy Reactive Logging
### Description

#31975 introduced a debug log statement for `WebApplicationException`s thrown during RESTEasy Reactive requests. I would like to be able to configure the log level of that log statement with a configuration property.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by The-Huginn**
Resolves #31980

Currently supports only org.jboss.logging.Logger.Level as configuration.
Perhaps somebody can direct me to how I can accept JUL or jboss Level and convert them to jboss Level only?

### CODE REVIEW

**Review by xstefank — APPROVED** [review]
LGTM

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/spi/DefaultRuntimeConfiguration.java [hunk 1]
```diff
@@ -5,14 +5,20 @@
 import java.util.List;
 import java.util.Optional;
 
+import org.jboss.logging.Logger;
+
 public class DefaultRuntimeConfiguration implements RuntimeConfiguration {
     final Duration readTimeout;
+
+    final Logger.Level webApplicationExceptionLogLevel;
```
Comment (xstefank) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/spi/DefaultRuntimeConfiguration.java:13]: minor - why this needs to be package private?

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/spi/DefaultRuntimeConfiguration.java [hunk 2]
```diff
@@ -5,14 +5,20 @@
 import java.util.List;
 import java.util.Optional;
 
+import org.jboss.logging.Logger;
+
 public class DefaultRuntimeConfiguration implements RuntimeConfiguration {
     final Duration readTimeout;
```
Comment (The-Huginn) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/spi/DefaultRuntimeConfiguration.java:11]: to be consistent with with readTimeout attribute

**Review by Sgitario — COMMENTED** [review]
I'm a bit reluctant about this solution because:
- It adds a new property `quarkus.http.WebApplicationException.log.level` in a common extension (`vertx-http`) that is only used in Quarkus Resteasy Reactive. 
- It tries to resolve a logging problem that is already addressed using the standard Quarkus logging: https://quarkus.io/guides/logging

What I mean with the last statement is that if we want special logging handling for WebApplicationException exceptions, we could declare a new category in AbstractResteasyReactiveContext like:

```
protected static final Logger logExceptions = Logger.getLogger(WebApplicationException.class.getSimpleName());
```

Using this, we could now print these logs using `quarkus.log.category."WebApplicationException".level=DEBUG` (this should be documented). 

And additionally, we could even configure these traces to be printed using a file handler or any other trace handler. Which from my point of view is a much more powerful and lesser intrusive solution than the current approach. 

Having said that, I will @geoand the last words about it.

**Comment by geoand** [comment]
I agree with @Sgitario

**Comment by xstefank** [comment]
@The-Huginn this seems that it needs to be redone according to the @Sgitario's comment

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java [hunk 3]
```diff
@@ -31,6 +31,7 @@
 public class RuntimeExceptionMapper {
 
     private static final Logger log = Logger.getLogger(RuntimeExceptionMapper.class);
+    protected static final Logger logWebApplicationExceptions = Logger.getLogger(WebApplicationException.class.getSimpleName());
```
Comment (Sgitario) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/RuntimeExceptionMapper.java:34]: why protected ?

**Comment by Sgitario** [comment]
Apart from my comment, the rest of the changes look good to me. 
Up to @geoand to move forward.

**Comment by geoand** [comment]
I agree with the analysis you put forth in a previous comment

**Review by Sgitario — APPROVED** [review]

**Review by Sgitario — COMMENTED** [review]
The test failures are not related.
But I'm still missing changes to the documentation (in the RESTEasy Reactive guide) to describe to users how to use this.

File: docs/src/main/asciidoc/resteasy-reactive.adoc [hunk 4]
```diff
@@ -1975,6 +1975,16 @@ public class Endpoint {
 }
 ----
 
+[NOTE]
+====
+You can change the log level of the thrown `WebApplicationException` exceptions by configuring the following property `quarkus.http.WebApplicationException.log.level` like so:
+
+[source, properties]
+----
+quarkus.http.WebApplicationException.log.level=DEBUG
```
Comment (Sgitario) [inline docs/src/main/asciidoc/resteasy-reactive.adoc:1984]: This property does not exist. The right one I think it's `quarkus.log.category."WebApplicationException".level=DEBUG`

**Review by Sgitario — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/resteasy-reactive.adoc (hunk 4 above)
Comment (The-Huginn) [reply docs/src/main/asciidoc/resteasy-reactive.adoc]: Sorry for that, I didn't realize that

**Review by Sgitario — APPROVED** [review]

**Comment by Sgitario** [comment]
LGTM now. The test failures are unrelated. Merging.

**Pull request merged**